src/devices/acurite.c

```diff
@@ -490,55 +435,80 @@ static int acurite_txr_callback(bitbuffer_t *bitbuf) {
 
 	// The 5-n-1 weather sensor messages are 8 bytes.
 	if (browlen == ACURITE_5N1_BITLEN / 8) {
+        if (debug_output) {
+            fprintf(stderr, "Acurite 5n1 raw msg: %02X %02X %02X %02X %02X %02X %02X %02X\n",
+                bb[0], bb[1], bb[2], bb[3], bb[4], bb[5], bb[6], bb[7]);
+        }
 	    channel = acurite_getChannel(bb[0]);
+        sprintf(channel_str, "%c", channel);        
 	    sensor_id = acurite_5n1_getSensorId(bb[0],bb[1]);
 	    repeat_no = acurite_5n1_getMessageCaught(bb[0]);
 	    message_type = bb[2] & 0x3f;
-
+        battery_low = (bb[2] & 0x40) >> 6;
 
 	    if (message_type == 0x31) {
-		// Wind speed, wind direction, and rain fall
-	        wind_speed = acurite_getWindSpeed(bb[3], bb[4]);
-		wind_speedmph = kmph2mph(wind_speed);
-		wind_dird = acurite_5n1_winddirections[bb[4] & 0x0f];
-		wind_dirstr = acurite_5n1_winddirection_str[bb[4] & 0x0f];
-		raincounter = acurite_getRainfallCounter(bb[5], bb[6]);
-		if (acurite_5n1t_raincounter > 0) {
-		    // track rainfall difference after first run
-		    // FIXME when converting to structured output, just output
-		    // the reading, let consumer track state/wrap around, etc. 
-		    rainfall = ( raincounter - acurite_5n1t_raincounter ) * 0.01;
-		    if (raincounter < acurite_5n1t_raincounter) {
-			printf("%s Acurite 5n1 sensor 0x%04X Ch %c, rain counter reset or wrapped around (old %d, new %d)\n",
-			       time_str, sensor_id, channel, acurite_5n1t_raincounter, raincounter);
-			acurite_5n1t_raincounter = raincounter;
-		    }
-		} else {
-		    // capture starting counter
-		    acurite_5n1t_raincounter = raincounter;
-		    printf("%s Acurite 5n1 sensor 0x%04X Ch %c, Total rain fall since last reset: %0.2f\n",
-			   time_str, sensor_id, channel, raincounter * 0.01);
-		}
-
-		printf("%s Acurite 5n1 sensor 0x%04X Ch %c, Msg %02x, Wind %d kmph / %0.1f mph %0.1f° %s (%d), rain gauge %0.2f in.\n",
-		       time_str, sensor_id, channel, message_type,
-		       wind_speed, wind_speedmph,
-		       wind_dird, wind_dirstr, bb[4] & 0x0f, rainfall);
+            // Wind speed, wind direction, and rain fall
+            wind_speed = acurite_getWindSpeed_kph(bb[3], bb[4]);
+            wind_speedmph = kmph2mph(wind_speed);
+            wind_dird = acurite_5n1_winddirections[bb[4] & 0x0f];
+            wind_dirstr = acurite_5n1_winddirection_str[bb[4] & 0x0f];
+            raincounter = acurite_getRainfallCounter(bb[5], bb[6]);
+            if (acurite_5n1t_raincounter > 0) {
+                // track rainfall difference after first run
+                // FIXME when converting to structured output, just output
+                // the reading, let consumer track state/wrap around, etc. 
+                rainfall = ( raincounter - acurite_5n1t_raincounter ) * 0.01;
```

Comment:
most weather station hardware sends an accumulated value rather than a delta.  for some hardware, the accumulate resets each day, for others it resets when the hardware counter wraps around.

generally speaking, an accumulated value is less error-prone that deltas; if you are sending deltas and you lose a packet, you've lost data, if you're sending deltas and you send a packet twice, you have double-counted.

on the other hand, it sucks to maintain state.

in this case, i would recommend that the rtl_433 process maintain the accumulated state of the rain counter.  restart rtl_433, and the counter starts again.

any downstream consumer will have no problem, and treat it as a counter wraparound.